server/authenticators.py

```diff
@@ -1,114 +0,0 @@
-from flask import redirect
-from flask_oauthlib.client import OAuth
-from server.secret_keys import google_creds
-from collections import namedtuple
-
-import requests
-
-
-class AuthenticationException(Exception):
-    """
-    Exception thrown when authentication fails.
-    """
-    pass
-
-
-class Authenticator(object):
-    """
-    Authenticates a user with an access token.
-    """
-
-    def __init__(self, app):
-        pass
-
-    def authenticate(self, access_token):
-        """
-        Returns the email this access token corresponds to.
-        """
-        raise NotImplementedError
-
-
-class GoogleAuthenticator(Authenticator):
-    """
-    Authenticates a user with an access token using Google APIs.
-    """
-
-    API_URL = "https://www.googleapis.com/oauth2/v1/userinfo"
-
-    def __init__(self, app):
-        self.oauth = OAuth(app)
-        self.google = self.oauth.remote_app(
-            'google',
-            consumer_key=google_creds['GOOGLE_ID'],
-            consumer_secret=google_creds['GOOGLE_SECRET'],
-            request_token_params={
-                'scope': 'email'
-            },
-            base_url='https://www.googleapis.com/oauth2/v1/',
-            request_token_url=None,
-            access_token_method='POST',
-            access_token_url='https://accounts.google.com/o/oauth2/token',
-            authorize_url='https://accounts.google.com/o/oauth2/auth',
-        )
-
-    def authorize(self, callback):
-        return self.google.authorize(callback=callback)
-
-    def response(self):
-        return self.google.authorized_response()
-
-    def get(self, *args, **kwargs):
-        return self.google.get(*args, **kwargs)
-
-    def email(self, access_token):
-        return self.token_email(access_token)
-
-    @staticmethod
-    def token_email(access_token):
```

Comment:
The API relies on having a token_email function to call on all requests. We can either put this token_email function in there (along with a trapdoor for testing-login) or add it directly to the `auth.py` controller and import it in api. 

Any preference for where this goes? (I'm leaning towards the latter option)